dm/master/config_test.go

```diff
@@ -0,0 +1,186 @@
+		SampleConfigFile = encode
+
+		cfg := NewConfig()
+		err = cfg.Parse([]string{"-print-sample-config"})
+		c.Assert(err, check.ErrorMatches, flag.ErrHelp.Error())
+	})
+	c.Assert(strings.TrimSpace(out), check.Equals, strings.TrimSpace(string(buf)))
+
+	// test invalid base64 encoded sample config
+	out = capturer.CaptureStdout(func() {
+		SampleConfigFile = "invalid base64 encode string"
+
+		cfg := NewConfig()
+		err = cfg.Parse([]string{"-print-sample-config"})
+		c.Assert(err, check.ErrorMatches, flag.ErrHelp.Error())
+	})
+	c.Assert(strings.TrimSpace(out), check.Matches, "base64 decode config error:.*")
+}
+
+func (t *testMaster) TestConfig(c *check.C) {
+	var (
+		err        error
+		cfg        = &Config{}
+		masterAddr = ":8261"
+		deployMap  = map[string]string{
+			"mysql-replica-01": "172.16.10.72:8262",
+			"mysql-replica-02": "172.16.10.73:8262",
+		}
+		cases = []struct {
+			args     []string
+			hasError bool
+			errorReg string
+		}{
+			{
+				[]string{"-V"},
+				true,
+				flag.ErrHelp.Error(),
+			},
+			{
+				[]string{"-print-sample-config"},
+				true,
+				flag.ErrHelp.Error(),
+			},
+			{
+				[]string{"invalid"},
+				true,
+				"'invalid' is an invalid flag",
+			},
+			{
+				[]string{"--config=./dm-master.toml"},
+				false,
+				"",
+			},
+		}
+	)
+
+	cfg.ConfigFile = defaultConfigFile
+	err = cfg.Reload()
+	c.Assert(err, check.IsNil)
+	c.Assert(cfg.MasterAddr, check.Equals, masterAddr)
+	c.Assert(cfg.DeployMap, check.DeepEquals, deployMap)
+
+	for _, tc := range cases {
+		cfg = NewConfig()
+		err = cfg.Parse(tc.args)
+		if tc.hasError {
+			c.Assert(err, check.ErrorMatches, tc.errorReg)
+		} else {
+			c.Assert(cfg.MasterAddr, check.Equals, masterAddr)
+			c.Assert(cfg.DeployMap, check.DeepEquals, deployMap)
+			c.Assert(cfg.String(), check.Matches, fmt.Sprintf("{.*master-addr\":\"%s\".*}", masterAddr))
+		}
+	}
+}
+
+func (t *testMaster) TestUpdateConfig(c *check.C) {
+	var (
+		err        error
+		content    []byte
+		newContent []byte
+		newCfgPath string
+	)
+	newCfgPath = path.Join(c.MkDir(), "test_config.toml")
+	c.Assert(err, check.IsNil)
+
+	cfg := &Config{}
+	cfg.configFromFile(defaultConfigFile)
+
+	content, err = ioutil.ReadFile(defaultConfigFile)
+	c.Assert(err, check.IsNil)
+
+	// update config to a new file
+	cfg.ConfigFile = newCfgPath
+	err = cfg.UpdateConfigFile(string(content))
+	c.Assert(err, check.IsNil)
+	newContent, err = ioutil.ReadFile(newCfgPath)
+	c.Assert(err, check.IsNil)
+	c.Assert(newContent, check.DeepEquals, content)
+
+	// update config to the default config file
+	cfg.ConfigFile = ""
+	err = cfg.UpdateConfigFile(string(content))
+	c.Assert(err, check.IsNil)
+	newContent, err = ioutil.ReadFile(defaultConfigFile)
+	c.Assert(err, check.IsNil)
+	c.Assert(newContent, check.DeepEquals, content)
+}
+
+func (t *testMaster) TestInvalidConfig(c *check.C) {
+	var (
+		f   *os.File
+		err error
+		cfg = NewConfig()
+	)
+	f, err = ioutil.TempFile(".", "test_invalid_config.toml")
```

Comment:
how about use `:=` and don't need var f and err

Reply:
use `c.MkDir` and create file in temp dir instead